### ISSUE TITLE
Assigning own list unintentionally clears the list items.

### ISSUE THREAD

**Issue by zaki50** [realm/realm-java#5395]
#### Goal

`model.setColumnRealmList(model.getColumnRealmList());`  should be no-op.

#### Expected Results

The list keeps its contents.

#### Actual Results

The items in the list is cleared.

#### Steps & Code to Reproduce

``` java
    @Test
    public void assignOwnList() {
        RealmList<AllTypesRealmModel> allTypesRealmModels = new RealmList<AllTypesRealmModel>();
        AllTypesRealmModel allTypePojo;
        for (int i = 0; i < 2; i++) {
            allTypePojo = new AllTypesRealmModel();
            allTypePojo.columnLong = i;
            allTypesRealmModels.add(allTypePojo);
        }

        RealmModelWithRealmListOfRealmModel model = new RealmModelWithRealmListOfRealmModel();
        model.setColumnRealmList(allTypesRealmModels);

        realm.beginTransaction();
        model = realm.copyToRealm(model);
        realm.commitTransaction();

        assertEquals(2, model.getColumnRealmList().size());
        realm.beginTransaction();
        model.setColumnRealmList(model.getColumnRealmList());
        realm.commitTransaction();
        assertEquals(2, model.getColumnRealmList().size());
    }
```

#### Version of Realm and tooling
Realm version(s): 3.x

Realm sync feature enabled: no

Android Studio version: 3.0.0-beta7

Which Android version and device: any

**Comment by cmelchior** [comment]
Confirmed: https://github.com/realm/realm-java/pull/5396

**Comment by cmelchior** [comment]
It looks like we have the same problem for both `copyToRealmOrUpdate` and `insertOrUpdate`

**Comment by cmelchior** [comment]
Fixed in #5396
